script/server

```diff
@@ -6,5 +6,37 @@ set -e
 # check that we are in the expected directory
 cd `dirname $0`/..
 
-./manage.py migrate
-./manage.py runserver 0.0.0.0:8000
+ACTION=$1
+[ -n "$ACTION" ] || ACTION="--local"
+
+case "$ACTION" in
+    --local)
+        script/bootstrap
+        source .env
+        if [ "$DEVENV" == "docker" ] ; then
+            echo "==> Removing any stopped or crashed containers..."
+            docker-compose rm
+            echo "==> Removing node_modules volume if present..."
+            docker volume rm noiseworks_node_modules || true
+            echo "==> Starting development environment, hit ^C to stop..."
+            docker-compose up
```

Comment:
I tried this out and I got an error because of the django-environ upgrade - the container isn't rebuilt and I already had one, so I think it reuses that? Do we always have to rebuild just in case?

Reply:
So if we change the things in the *image* that we start a container from, then yes we need to rebuild the image - so that would include the Python and NPM dependencies in this case.

As long as you haven't removed your local Docker cache, rebuilding an image that has no changes will simply use the cached layers in almost all cases so it shouldn't be that time consuming to add a call to build a new image each time, or you can manually run a build when you change something that you know needs it like a dependency package (which I usually do as a test anyway).

It's a bit of a balance finding a development workflow that requires absolutely no additional manual interaction and running things like a build set "just in case"...

Reply:
Yeah, I see what you mean - I wonder if you can flag in a commit (like my django-environ commit) that it means a rebuild is necessary on next run, hmm!